x/sync/network_client.go

```diff
@@ -186,11 +186,6 @@ func (c *networkClient) RequestAny(
 // If the limit on active requests is reached, this function blocks until
 // a slot becomes available.
 func (c *networkClient) Request(ctx context.Context, nodeID ids.NodeID, request []byte) ([]byte, error) {
-	// TODO danlaine: is it possible for this condition to occur?
-	if nodeID == ids.EmptyNodeID {
-		return nil, fmt.Errorf("cannot send request to empty nodeID, nodeID=%s, requestLen=%d", nodeID, len(request))
```

Comment:
This seems like a useful check to leave on the exported method, I think.
Even if all current uses are correct it might catch errors in future development.

Not very strongly opinionated, happy to defer to @dboehm-avalabs